youtube_dl/extractor/liveleak.py

```diff
@@ -120,6 +120,19 @@ def _real_extract(self, url):
             }
 
         for idx, info_dict in enumerate(entries):
+
+            for f in info_dict['formats']:
+                match = re.match(r'https?://(?:\w+\.)?liveleak\.com/.*?\.(\w+)(\.\w+\.mp4)(?:\?.+)?', f['url'])
```

Comment:
`re.sub`.

Reply:
I'm using results of the match groups here, how would I do this with `re.sub`?

Reply:
By read how `re.sub` works in the first place?

Reply:
I did read into it, but as I don't want to replace the first group (which is the original file extension), I don't see how `re.sub` would give me the replaced string *and* the content of the group. Perhaps I'm overlooking it?

Reply:
Don't replace it then. What's the problem?

Reply:
Would you prefer to use `re.sub` for the replacement and then `re.match` to grab the extension?

Reply:
No. You don't need to grab any extension in the first place.

Reply:
Isn't that what the `ext` is meant for?

Reply:
Again: you should not bother with that. It will be automatically extracted from the download URL.

Reply:
Ah, that wasn't clear to me.